smtp/message.js

```diff
@@ -309,34 +441,38 @@ class MessageStream extends Stream {
 			fs.open(attachment.path, 'r', opened);
 		};
 
+		/**
+		 * @param {MessageAttachment} attachment the metadata to use as headers
+		 * @param {function(): void} callback the function to call after output is finished
+		 * @returns {void}
+		 */
 		const output_stream = (attachment, callback) => {
 			if (attachment.stream.readable) {
-				let previous = null;
+				let previous = Buffer.alloc(0);
 
 				attachment.stream.resume();
 
 				attachment.stream.on('end', () => {
-					output_base64(
-						(previous || Buffer.from(0)).toString('base64'),
-						callback
-					);
+					output_base64(previous.toString('base64'), callback);
 					this.removeListener('pause', attachment.stream.pause);
 					this.removeListener('resume', attachment.stream.resume);
 					this.removeListener('error', attachment.stream.resume);
 				});
 
-				attachment.stream.on('data', buffer => {
+				attachment.stream.on('data', buff => {
 					// do we have bytes from a previous stream data event?
-					if (previous) {
+					let buffer = Buffer.isBuffer(buff) ? buff : Buffer.from(buff);
+
+					if (previous.byteLength > 0) {
 						const buffer2 = Buffer.concat([previous, buffer]);
-						previous = null; // free up the buffer
+						previous = Buffer.alloc(0); // free up the buffer
 						buffer = null; // free up the buffer
 						buffer = buffer2;
 					}
 
 					const padded = buffer.length % MIME64CHUNK;
 					// encode as much of the buffer to base64 without empty bytes
-					if (padded) {
+					if (padded > 0) {
```

Comment:
since buffer length has a minimum value of 0, it should be fine to opt for the more efficient check

Reply:
agreed.